src/websocket.cpp

```diff
@@ -13,13 +13,14 @@
 
 #include "impl/internals.hpp"
 #include "impl/websocket.hpp"
+#include "impl/certificate.hpp"
 
 namespace rtc {
 
 WebSocket::WebSocket() : WebSocket(Configuration()) {}
 
-WebSocket::WebSocket(Configuration config)
-    : CheshireCat<impl::WebSocket>(std::move(config)),
+WebSocket::WebSocket(const Configuration& config)
```

Comment:
I see why you need such a change, but I'm not a fan of changing the signature here, because it breaks consistency with the rest of the API and prevents from moving the configuration all the way by forcing a copy when constructing `impl::WebSocket`.

Could you please move the content of `loadCertificate()` to the body of the constructor of `impl::WebSocket` instead? You can check if `certificate` is null there and attempt to load from `config` if so.

Reply:
Sure. I moved the whole helper to preserve `mCertificate` constness.